Modules/Tests/YosemiteTests/Storage/PersistedProductTests.swift

```diff
@@ -0,0 +1,412 @@
+            shortDescription: "ShortD",
+            sku: nil,
+            globalUniqueID: nil,
+            price: "12.34",
+            downloadable: true,
+            parentID: 0,
+            manageStock: false,
+            stockQuantity: nil,
+            stockStatusKey: "outofstock"
+        )
+
+        let productImages = [
+            PersistedProductImage(id: 200,
+                                  productID: productID,
+                                  dateCreated: Date(timeIntervalSince1970: 10),
+                                  dateModified: nil,
+                                  src: "https://example.com/p1.png",
+                                  name: "p1",
+                                  alt: "a1"),
+            PersistedProductImage(id: 201,
+                                  productID: productID,
+                                  dateCreated: Date(timeIntervalSince1970: 11),
+                                  dateModified: Date(timeIntervalSince1970: 12),
+                                  src: "https://example.com/p2.png",
+                                  name: nil,
+                                  alt: nil)
+        ]
+
+        let persistedAttributes = [
+            PersistedProductAttribute(productID: productID, name: "Material", position: 0, visible: true, variation: false, options: ["Cotton"]),
+            PersistedProductAttribute(productID: productID, name: "Fit", position: 1, visible: true, variation: true, options: ["Slim", "Regular"])
+        ]
+
+        // When
+        let posProduct = persisted.toPOSProduct(
+            images: productImages.map { $0.toProductImage() },
+            attributes: persistedAttributes.map { $0.toProductAttribute(siteID: siteID) }
+        )
+
+        // Then
+        #expect(posProduct.siteID == siteID)
+        #expect(posProduct.productID == productID)
+        #expect(posProduct.name == persisted.name)
+        #expect(posProduct.productTypeKey == persisted.productTypeKey)
+        #expect(posProduct.fullDescription == persisted.fullDescription)
+        #expect(posProduct.shortDescription == persisted.shortDescription)
+        #expect(posProduct.sku == persisted.sku)
+        #expect(posProduct.globalUniqueID == persisted.globalUniqueID)
+        #expect(posProduct.price == persisted.price)
+        #expect(posProduct.downloadable == persisted.downloadable)
+        #expect(posProduct.parentID == persisted.parentID)
+        #expect(posProduct.manageStock == persisted.manageStock)
+        #expect(posProduct.stockQuantity == persisted.stockQuantity)
+        #expect(posProduct.stockStatusKey == persisted.stockStatusKey)
+        #expect(posProduct.images.count == 2)
+        #expect(posProduct.attributes.count == 2)
+        #expect(posProduct.attributesForVariations.count == 1)
+    }
+
+    @Test("A Product's images and attributes are fetched automatically")
+    func test_product_with_associations_fetches_related_records() throws {
```

Comment:
super nit: can add Given/When/Then for consistency (same for `test_product_without_related_records_creates_empty_arrays`).